samples/javascript_nodejs/21.corebot-app-insights/dialogs/mainDialog.js

```diff
@@ -0,0 +1,169 @@
+            this.finalStep.bind(this)
+        ]);
+
+        // Define the main dialog and its related components.
+        // This is a sample "book a flight" dialog.
+        this.addDialog(textPrompt)
+            .addDialog(bookingDialog)
+            .addDialog(waterfallDialog);
+
+        this.initialDialogId = MAIN_WATERFALL_DIALOG;
+    }
+
+    /**
+     * The run method handles the incoming activity (in the form of a TurnContext) and passes it through the dialog system.
+     * If no dialog is active, it will start the default dialog.
+     * @param {*} turnContext
+     * @param {*} accessor
+     */
+    async run(turnContext, accessor) {
+        const dialogSet = new DialogSet(accessor);
+        dialogSet.add(this);
+
+        const dialogContext = await dialogSet.createContext(turnContext);
+        const results = await dialogContext.continueDialog();
+        if (results.status === DialogTurnStatus.empty) {
+            await dialogContext.beginDialog(this.id);
+        }
+    }
+
+    /**
+     * First step in the waterfall dialog. Prompts the user for a command.
+     * Currently, this expects a booking request, like "book me a flight from Paris to Berlin on march 22"
+     * Note that the sample LUIS model will only recognize Paris, Berlin, New York and London as airport cities.
+     */
+    async introStep(stepContext) {
+        if (!this.luisRecognizer.isConfigured) {
+            const messageText = 'NOTE: LUIS is not configured. To enable all capabilities, add `LuisAppId`, `LuisAPIKey` and `LuisAPIHostName` to the .env file.';
+            await stepContext.context.sendActivity(messageText, null, InputHints.IgnoringInput);
+            return await stepContext.next();
+        }
+
+        const messageText = stepContext.options.restartMsg ? stepContext.options.restartMsg : 'What can I help you with today?\nSay something like "Book a flight from Paris to Berlin on March 22, 2020"';
+        const promptMessage = MessageFactory.text(messageText, messageText, InputHints.ExpectingInput);
+        return await stepContext.prompt('TextPrompt', { prompt: promptMessage });
+    }
+
+    /**
+     * Second step in the waterfall.  This will use LUIS to attempt to extract the origin, destination and travel dates.
+     * Then, it hands off to the bookingDialog child dialog to collect any remaining details.
+     */
+    async actStep(stepContext) {
+        const bookingDetails = {};
+
+        if (!this.luisRecognizer.isConfigured) {
+            // LUIS is not configured, we just run the BookingDialog path.
+            return await stepContext.beginDialog('bookingDialog', bookingDetails);
+        }
+
+        // Call LUIS and gather any potential booking details. (Note the TurnContext has the response to the prompt)
+        const luisResult = await this.luisRecognizer.executeLuisQuery(stepContext.context);
+        switch (LuisRecognizer.topIntent(luisResult)) {
+        case 'BookFlight': {
```

Comment:
the brackets  '{' are not required in general, and not needed to implement telemetry into the coreBot sample. please remove.
```suggestion
        case 'BookFlight': 
```